### ISSUE TITLE
Question about contributing to filters

### ISSUE THREAD

**Issue by K2DLS** [fail2ban/fail2ban#1670]
I've been making some extensive use of fail2ban and have been doing a lot of filter tuning.  I want to give back to the project, so rather than just file reports, I've contributed some filter enhancements for sshd.conf.  Others will follow.

I noticed that my pull request has a red "x" next to it with the notation "The Travis CI build failed."  I'm wondering what this means?  Could someone better versed in the project point this "noob" in the right direction?  The pull request in question is #1662.

Thanks!

**Comment by M66B** [comment]
I have the same question and motive as @K2DLS (contributing back)

Here are some WordPress filters which could be useful to others as well:
https://www.lowendtalk.com/discussion/comment/2021256/#Comment_2021256

What is the correct way to submit those filters?

**Comment by sebres** [comment]
Fail2ban is test-cases covered. So each PR will be validated to pass all the test-cases.

If you look at https://travis-ci.org/fail2ban/fail2ban/jobs/191072817#L640 you'll see ...
```
AssertionError: Line not matched when should have: /home/travis/build/fail2ban/fail2ban/fail2ban/tests/files/logs/sshd:145 'Nov 23 21:50:37 sshd[8148]: Connection closed by 61.0.0.1 [preauth]\n'
```

AS related to filters: Several test-cases cover our filters, especially each failregex should have at least one line (described within inline JSON-comment) in the test-file in `fail2ban/tests/files/logs/`, that this regex matches.

For sshd-filter it is https://github.com/fail2ban/fail2ban/blob/0.10/fail2ban/tests/files/logs/sshd .

If you'll extend a filter you should add a new entries there. If you'll remove something from the filter you should also remove corresponding entries there.

You can also test it local on your computer (don't need wait for processing the PR in travis after each commit):
```bash
cd my-fail2ban-working-copy
# for help:
PYTHONPATH=. bin/fail2ban-testcases --help
# test sshd filter only:
PYTHONPATH=. bin/fail2ban-testcases SSHD
# fast complete test, without network, etc:
PYTHONPATH=. bin/fail2ban-testcases -fng
# complete test:
PYTHONPATH=. bin/fail2ban-testcases
```
